### PR TITLE
Reacting to Hosting changes

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @JunTaoLuo \ @davidfowl

**Comment by pranavkm** [comment]
cc @NTaylorMullen

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs [hunk 1]
```diff
@@ -20,21 +17,21 @@ public class RazorViewEngineOptionsSetup : ConfigureOptions<RazorViewEngineOptio
...
         {
         }
 
         private static void ConfigureRazor(
             RazorViewEngineOptions razorOptions,
-            IApplicationEnvironment applicationEnvironment,
             IHostingEnvironment hostingEnvironment)
         {
-            razorOptions.FileProviders.Add(new PhysicalFileProvider(applicationEnvironment.ApplicationBasePath));
+            if (hostingEnvironment.ContentRootFileProvider != null)
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs]: This is not really necessary, ContentRootFileProvider is always set.

**Comment by JunTaoLuo** [comment]
Looks good, :shipit: if the tests pass locally. I'm assuming the travis and appveyor tests are failing due to old packages or some other known issue.

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs]: It's settable though, isn't it?

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptionsSetup.cs]: Makes sense.

**Pull request closed without merging**